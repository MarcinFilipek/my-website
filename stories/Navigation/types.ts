type NavItem = {
  id: number;
  text: string;
  href: string;
};

export type NavItems = NavItem[];
