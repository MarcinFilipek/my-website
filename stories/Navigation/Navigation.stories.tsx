import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import { Navigation } from "./Navigation";

const Template: ComponentStory<typeof Navigation> = (args) => (
  <div
    style={{
      width: "500px",
      height: "100%",
      padding: "20px",
    }}
  >
    <Navigation {...args} />
  </div>
);

export const Default = Template.bind({});

Default.args = {
  navItems: [
    { id: 0, text: "Home", href: "#" },
    { id: 0, text: "Blog", href: "#" },
    { id: 0, text: "O mnie", href: "#" },
  ],
};

export default {
  title: "Components/Navigation",
  component: Navigation,
} as ComponentMeta<typeof Navigation>;
