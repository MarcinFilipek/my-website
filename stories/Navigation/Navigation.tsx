import React from "react";
import { Typography, useTheme } from "@mui/material";
import { NavItems } from "./types";
import Link from "next/link";

type Props = {
  navItems: NavItems;
};

export const Navigation = ({ navItems }: Props) => {
  const theme = useTheme();

  return (
    <nav
      style={{
        border: `1px solid ${theme.palette.primary.main}66`,
        background: "#100d21",
        borderRadius: "0.8rem",
        display: "inline-block",
        padding: "20px",
      }}
      data-testid="navigation"
    >
      <ul
        style={{
          display: "flex",
          listStyle: "none",
          gap: "50px",
          padding: "0",
          margin: "0",
        }}
      >
        {navItems.map((navItem) => (
          <li key={navItem.id}>
            <Link href={navItem.href} style={{ textDecoration: "none" }}>
              <Typography
                variant="body1"
                style={{ color: theme.palette.primary.light }}
                data-testid={`nav-item-${navItem.id}`}
              >
                {navItem.text}
              </Typography>
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};
