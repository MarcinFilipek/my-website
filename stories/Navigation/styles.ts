import { styled } from "@mui/material/styles";
import Link from "next/link";
import { Typography } from "@mui/material";

export const Container = styled("nav")(
  ({ theme }) => `
        border: 1px solid ${theme.palette.primary.main}66;
        background: #100d21;
        border-radius: 0.8rem;
        display: inline-block;
        padding: 20px;`
);

export const StyledUl = styled("ul")(
  ({ theme }) => `
        display: flex;
        list-style: none;
        gap: 50px;
        padding: 0;
        margin: 0;`
);

export const StyledLink = styled(Link)(
  ({ theme }) => `
  text-decoration: none;`
);

export const StyledTypography = styled(Typography)(
  ({ theme }) => `color: ${theme.palette.primary.light};`
);
