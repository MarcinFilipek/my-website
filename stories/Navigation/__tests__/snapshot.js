import { render } from "@testing-library/react";
import { Navigation } from "../Navigation";

it("renders homepage unchanged", () => {
  const navItems = [
    { id: 0, text: "Home", href: "#" },
    { id: 1, text: "Blog", href: "#" },
    { id: 2, text: "About", href: "#" },
  ];
  const { container } = render(<Navigation navItems={navItems} />);
  expect(container).toMatchSnapshot();
});
