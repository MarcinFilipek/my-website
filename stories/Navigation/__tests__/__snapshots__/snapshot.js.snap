// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <nav
    class="css-15bfjca"
    data-testid="navigation"
  >
    <ul
      class="css-184no2y"
    >
      <li>
        <p
          class="MuiTypography-root MuiTypography-body1 css-159rp9v-MuiTypography-root"
          data-testid="nav-item-0"
        >
          Home
        </p>
      </li>
      <li>
        <p
          class="MuiTypography-root MuiTypography-body1 css-159rp9v-MuiTypography-root"
          data-testid="nav-item-1"
        >
          Blog
        </p>
      </li>
      <li>
        <p
          class="MuiTypography-root MuiTypography-body1 css-159rp9v-MuiTypography-root"
          data-testid="nav-item-2"
        >
          About
        </p>
      </li>
    </ul>
  </nav>
</div>
`;
